components/selection/selection-mixin.js

```diff
@@ -74,7 +95,40 @@ export const SelectionMixin = superclass => class extends superclass {
 		this._updateSelectionObservers();
 	}
 
-	_handleSelectionChange() {
+	_handleRadioKeyDown(e) {
+		// check composed path for radio (e.target could be d2l-list-item or other element due to retargeting)
+		if (!e.composedPath()[0].classList.contains('d2l-selection-input-radio')) return;
+		if (e.keyCode >= keyCodes.LEFT && e.keyCode <= keyCodes.DOWN) e.preventDefault();
+	}
+
+	_handleRadioKeyUp(e) {
+		// check composed path for radio (e.target could be d2l-list-item or other element due to retargeting)
+		if (!e.composedPath()[0].classList.contains('d2l-selection-input-radio')) return;
+		if (e.keyCode < keyCodes.LEFT || e.keyCode > keyCodes.DOWN) return;
+
+		const selectables = Array.from(this._selectionSelectables.values());
+		let currentIndex = selectables.findIndex(selectable => selectable.selected);
+		if (currentIndex === -1) currentIndex = 0;
+		let newIndex;
+
+		if (e.keyCode === keyCodes.RIGHT || e.keyCode === keyCodes.DOWN) {
```

Comment:
This effectively mimics the native radio behaviour of changing the selection using arrow keys.

Reply:
Do we need to invert anything for RTL?

Reply:
I'll check and match to the behaviour for native radios in an RTL context.